htmresearch/algorithms/apical_dependent_temporal_memory.py

```diff
@@ -419,6 +442,57 @@ def _calculateSegmentActivity(connections, activeInput, connectedPermanence,
             matchingSegments,
             potentialOverlaps)
 
+  @staticmethod
+  def _calculateBasalSegmentActivity(connections, activeInput,
```

Comment:
You could do all of this with a single `_calculateSegmentActivity` method. Just have a `reducedThresholdCells` parameter on the method. When we use it to calculate active apical segments, just pass in `()` for the `reducedThresholdCells`.